src/test/testing_assert.cpp

```diff
@@ -42,18 +56,65 @@ Matrix _table_to_matrix(const std::shared_ptr<const opossum::Table>& t) {
   return matrix;
 }
 
-std::string _matrix_to_string(const std::string& title, const Matrix& m) {
+std::string _matrix_to_string(const Matrix& matrix, const std::vector<std::pair<uint64_t, uint16_t>>& highlight_cells,
+                              const std::string& highlight_color) {
   std::stringstream stream;
-  stream << "-------" << title << "-------" << std::endl;
-  for (unsigned row = 0; row < m.size(); row++) {
-    for (opossum::ColumnID col{0}; col < m[row].size(); col++) {
-      stream << std::setw(8) << m[row][col] << " ";
+  bool highlight;
+  bool previous_row_highlighted = false;
+
+  std::string highlight_color_bg = ANSI_COLOR_BG_RED;
+  if (highlight_color == ANSI_COLOR_GREEN) {
+    highlight_color_bg = ANSI_COLOR_BG_GREEN;
+  }
+
+  for (unsigned row = 0; row < matrix.size(); row++) {
+    highlight = false;
+    auto it = std::find_if(highlight_cells.begin(), highlight_cells.end(),
+                           [&](const std::pair<uint64_t, uint16_t>& element) { return element.first == row; });
+    if (it != highlight_cells.end()) {
+      highlight = true;
+      if (!previous_row_highlighted) {
+        stream << "<<<<<" << std::endl;
+        previous_row_highlighted = true;
+      }
+    } else {
+      previous_row_highlighted = false;
+    }
+
+    // Highlight row number with background color
+    std::string coloring = "";
```

Comment:
No need to initialize std::string

Reply:
I prefer `auto coloring = std::string{}` or `std::string coloring{}`;

Reply:
Done.